### PR TITLE
Define `task_weights` if it is `None` in `MulticlassClassificationFFN`

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
The loss function requires `task_weights` as an attribute. If not provided, it is automatically initialized as `task_weights = torch.ones(n_tasks)` in `_FFNPredictorBase`. However, `n_tasks` is overwritten as `n_tasks * n_classes` for `MulticlassClassificationFFN`, leading to incorrect dimensions. I rebuild the criterion object with correct `task_weights ` in `MulticlassClassificationFFN` to solve this bug.

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

**Review by KnathanM — COMMENTED** [review]
Would it also work to do `task_weights = torch.ones(n_tasks) if task_weights is None else task_weights` before the call to `super().__init__()` on line 263? Then you wouldn't need to recreate the criterion.

**Comment by shihchengli** [comment]
@KnathanM It would work. Thanks for the suggestion.

**Comment by KnathanM** [comment]
I see that in `_FFNPredictorBase`, we have
```
    @property
    def n_tasks(self) -> int:
        return self.output_dim // self.n_targets
```

This will be incorrect in `MulticlassClassificationFFN` for the same reason as the task weights being incorrect. Do you think it is in the scope of this PR to also overwrite this, maybe by adding something like this to the class?
```
    @property
    def n_tasks(self) -> int:
        return self.output_dim // (self.n_targets * self.n_classes)
```

**Comment by shihchengli** [comment]
That is a good catch. I think it doesn't hurt to include it in this PR as well.

**Review by KnathanM — APPROVED** [review]
